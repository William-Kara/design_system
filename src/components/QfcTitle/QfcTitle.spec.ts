import { it, describe, beforeEach, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Title from './QfcTitle.vue';

let wrapper: any = null;

const propsValues = {
  text: 'Title',
};

beforeEach(() => {
  wrapper = shallowMount(Title, {
    propsData: propsValues,
  });
});

describe('Title Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should display a div title', () => {
    expect(wrapper.find('div').exists()).toBe(true);
  });
  it('should display a h1 title', () => {
    wrapper = shallowMount(Title, {
      propsData: {
        ...propsValues,
        level: 1,
      },
    });
    expect(wrapper.find('h1').exists()).toBe(true);
  });
  it('should display a h3 title', () => {
    wrapper = shallowMount(Title, {
      propsData: {
        ...propsValues,
        level: 3,
      },
    });
    expect(wrapper.find('h3').exists()).toBe(true);
  });
});
