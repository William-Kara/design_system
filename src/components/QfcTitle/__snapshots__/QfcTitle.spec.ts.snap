// Vitest Snapshot v1

exports[`Title Component > should render correctly 1`] = `"<div data-v-1c9f9ff3=\\"\\" class=\\"qfc-title\\">Title</div>"`;
